plugins/authentication/src/native.cpp

```diff
@@ -378,8 +377,11 @@ namespace irods
                                 {"scheme", scheme_name}, {irods_auth::next_operation, AUTH_CLIENT_START}};
                         }
                     }
-                    catch (const boost::bad_lexical_cast& e) {
-                        THROW(SYS_INVALID_INPUT_PARAM, fmt::format("invalid TTL [{}]", ttl_str));
+                    catch (const std::invalid_argument& e) {
+                        THROW(SYS_INVALID_INPUT_PARAM, fmt::format("Invalid TTL [{}]: {}", ttl_str, e.what()));
+                    }
+                    catch (const std::out_of_range& e) {
+                        THROW(SYS_INVALID_INPUT_PARAM, fmt::format("Invalid TTL [{}]: {}", ttl_str, e.what()));
```

Comment:
I take it the lack of a catch block for `std::exception` is intentional as it is handled by an earlier frame in the callstack?

The use of `THROW` near the end of this function hints at that.

Reply:
The `std::exception` would eventually be caught by `authenticate_client`, yes. We can handle it here, too, if that's preferred, though.

Reply:
No, I was just wondering. It's fine as is.